### DISCUSSION TITLE
useRealtime join?

### DISCUSSION THREAD

**Discussion by bruceharrison1984** [tmm/react-supabase#32] [General]
When utilizing `useRealtime` with a select clause, it is not applied when new data comes in.

For example:
```js
  const [calendarEvents, fetchCalendarEvents] = useRealtime<CalendarEvent>(
    'calendar_event',
    {
      select: {
        columns: '*, owner: user_profile(full_name, avatar)',
      },
    }
  );
```

The first usage of `calendarEvent` will have the joined data. Inserting a new item in to `calendar_event` will trigger a fetch and return the new `calendarEvent` data, but the `owner` property on the new data is `null`. Refreshing the page correctly pulls all data.

Am i missing something here, or is this intended/expected behavior?

**Comment by tmm** [comment]
This is the expected behavior right now based on how [Supabase subscriptions](https://supabase.com/docs/reference/javascript/subscribe) work.

There might be a clever way to detect that a initial query has relations and auto-fetch them on subscription changes.

**Reply by bruceharrison1984** [reply]
ok, thanks! It's easy enough to create a junction object that does the join anytime the realtime subscription updates.
